platform/ios/src/MGLMapView.mm

```diff
@@ -6098,7 +6103,19 @@ - (void)locationManagerDidChangeAuthorization:(id<MGLLocationManager>)manager
         [self.locationManager stopUpdatingLocation];
         [self.locationManager stopUpdatingHeading];
     } else {
-        [self validateLocationServices];
+        if (@available(iOS 14, *)) {
+            if ((manager.authorizationStatus != kCLAuthorizationStatusRestricted ||
+                 manager.authorizationStatus != kCLAuthorizationStatusAuthorizedAlways ||
+                 manager.authorizationStatus != kCLAuthorizationStatusAuthorizedWhenInUse) &&
+                manager.accuracyAuthorization == CLAccuracyAuthorizationReducedAccuracy &&
+                [self accuracyDescriptionString] != nil ) {
+                [self.locationManager requestTemporaryFullAccuracyAuthorizationWithPurposeKey:@"MBXAuthorizationAccuracyDescription"];
+            } else {
+                [self validateLocationServices];
+            }
```

Comment:
Do we need a `#if __IPHONE_OS_VERSION_MAX_ALLOWED >= 140000` here?

Reply:
Yes.